internal/mode/static/state/graph/nginxproxy.go

```diff
@@ -126,5 +127,56 @@ func validateNginxProxy(
 		npCfg.Spec.IPFamily = helpers.GetPointer[ngfAPI.IPFamilyType](ngfAPI.Dual)
 	}
 
+	allErrs = append(allErrs, validateRewriteClientIP(npCfg)...)
+
+	return allErrs
+}
+
+func validateRewriteClientIP(npCfg *ngfAPI.NginxProxy) field.ErrorList {
+	var allErrs field.ErrorList
+	spec := field.NewPath("spec")
+
+	if npCfg.Spec.RewriteClientIP != nil {
+		rewriteClientIP := npCfg.Spec.RewriteClientIP
+		rewriteClientIPPath := spec.Child("rewriteClientIP")
+		trustedAddressesPath := rewriteClientIPPath.Child("trustedAddresses")
+
+		if rewriteClientIP.Mode != nil {
+			mode := *rewriteClientIP.Mode
+			if len(rewriteClientIP.TrustedAddresses) == 0 {
+				allErrs = append(
+					allErrs,
+					field.Required(rewriteClientIPPath, "trustedAddresses field required when mode is set"),
+				)
+			}
+
+			switch mode {
+			case ngfAPI.RewriteClientIPModeProxyProtocol, ngfAPI.RewriteClientIPModeXForwardedFor:
+			default:
+				allErrs = append(
+					allErrs,
+					field.NotSupported(
+						rewriteClientIPPath.Child("mode"),
+						mode,
+						[]string{string(ngfAPI.RewriteClientIPModeProxyProtocol), string(ngfAPI.RewriteClientIPModeXForwardedFor)},
+					),
+				)
+			}
+		}
+
+		if len(rewriteClientIP.TrustedAddresses) > 16 {
+			allErrs = append(
+				allErrs,
+				field.TooLongMaxLength(trustedAddressesPath, rewriteClientIP.TrustedAddresses, 16),
+			)
+		}
+
+		for _, addr := range rewriteClientIP.TrustedAddresses {
+			if err := k8svalidation.IsValidCIDR(trustedAddressesPath, string(addr)); err != nil {
```

Comment:
Does this allow IP addresses?

Reply:
No, only allows valid [CIDRs](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation#IsValidCIDR)